From f6ffe2169c3dc02844b115821b3e79497aa1f68d Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 7 Nov 2018 22:19:07 +0800
Subject: [PATCH 5045/5242] MLK-20205-1: dmaengine: fsl-edma-v3: fix NULL
 pointer dereference

commit  c7c3d861c4da9ab943e091ec8656a540ded68302 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix 'null pointer dereferences issue' reported by coverity(CID-1477441).

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/fsl-edma-v3.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/fsl-edma-v3.c b/drivers/dma/fsl-edma-v3.c
index 47c7a17..6badda9 100644
--- a/drivers/dma/fsl-edma-v3.c
+++ b/drivers/dma/fsl-edma-v3.c
@@ -422,7 +422,7 @@ static enum dma_status fsl_edma3_tx_status(struct dma_chan *chan,
 	if (fsl_chan->edesc && cookie == fsl_chan->edesc->vdesc.tx.cookie)
 		txstate->residue = fsl_edma3_desc_residue(fsl_chan, vdesc,
 								true);
-	else if (vdesc)
+	else if (fsl_chan->edesc && vdesc)
 		txstate->residue = fsl_edma3_desc_residue(fsl_chan, vdesc,
 								false);
 	else
-- 
1.7.9.5

